front/components/actions/websearch/utils.ts

```diff
@@ -8,16 +8,16 @@ interface WebsearchResultCitation {
 }
 
 export function makeWebsearchResultsCitations(
-  action: WebsearchActionType
+  results: WebsearchResultType[]
 ): WebsearchResultCitation[] {
-  return (
-    action.output?.results.map((r) => {
-      return {
-        description: r.snippet,
-        href: r.link,
-        title: r.title,
-        type: "document",
-      };
-    }) ?? []
-  );
+  return results.map((r) => {
+    return {
+      description: r.snippet,
+      // @todo[daph] 2024-07-19 Remove the fallback on link.
+      // @ts-expect-error Property 'link' does not exist on type 'WebsearchResultType'.
+      href: r.sourceUrl ?? r.link,
```

Comment:
Can we keep link as optional in the type instead?

Reply:
Added, with a comment to make sure I cleanup